### PR TITLE
Add SNAP minimum benefit

### PR DESCRIPTION
**Pull request by nmrodelo**
Draft/ work in progress pull request for snap minimum benefit parameters

### CODE REVIEW

File: openfisca_us/parameters/benefit/snap/minimum_benefit.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+description:  Minimum SNAP Benefit by Geography and Household Size
+
+main:
```
Comment (MaxGhenis) [inline openfisca_us/parameters/benefit/snap/minimum_benefit.yaml:3]: What will `main` distinguish from?

File: openfisca_us/parameters/benefit/snap/minimum_benefit.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    1:
+      2021-01-01: snap.amount.main.1 * 0.08
+    2:
+      2021-01-01: snap.amount.main.1 * 0.08
+
+metadata:
+  unit: currency-USD
+  reference:
+  - title: '7 CFR § 273.10 - Determining household eligibility and benefit levels.'
+    href: https://www.law.cornell.edu/cfr/text/7/273.10  period: month
```
Comment (MaxGhenis) [inline openfisca_us/parameters/benefit/snap/minimum_benefit.yaml:44]:
```suggestion
    href: https://www.law.cornell.edu/cfr/text/7/273.10
```
Proportions won't need periods.

Comment (MaxGhenis) [inline openfisca_us/parameters/benefit/snap/minimum_benefit.yaml:44]: Not seeing 8% here

**Review by MaxGhenis — COMMENTED** [review]
I know just a draft but a couple comments after checking it out

File: openfisca_us/parameters/benefit/snap/minimum_benefit.yaml (hunk 1 above)
Comment (nmrodelo) [reply openfisca_us/parameters/benefit/snap/minimum_benefit.yaml]: nothing, just removed it

**Comment by MaxGhenis** [comment]
I'm not sure why `relevant_max_benefit` is null in the calculation. @nikhilwoodruff any ideas?

https://github.com/PolicyEngine/openfisca-us/runs/4419215730?check_suite_focus=true#step:5:591

**Comment by nmrodelo** [comment]
@nikhilwoodruff could you take a look at this?

File: openfisca_us/variables/usda/snap.py [hunk 3]
```diff
@@ -5,6 +5,32 @@
...
+
+    def formula(spm_unit, period, parameters):
+        # Parameters for the minimum benefit.
+        snap = parameters(period).usda.snap
+        min_benefit = snap.minimum_benefit
+        # Calculate the relevant maximum benefit, defined as the maximum
+        # benefit for a household of a certain size in their state.
+        state_group = spm_unit.household("state_group_str", period)
+        relevant_max_benefit = snap.amount.main[state_group][
+            min_benefit.relevant_max_benefit_household_size
```
Comment (nikhilwoodruff) [inline openfisca_us/variables/usda/snap.py:22]:
```suggestion
            str(min_benefit.relevant_max_benefit_household_size)
```

**Review by nikhilwoodruff — COMMENTED** [review]
Just took a look @nmrodelo - it's what I think is a bug in Core (I've seen it before in a few places). Essentially, when we index a parameter with an array of numbers (e.g. amount_by_household_size[[1, 1, 1, 3, 1, 2]]) we get the right values, but when we index by just a single number (amount_by_household_size[1]) it doesn't seem to work - except when we do amount_by_household_size["1"]. Here's how I fixed it locally below.

**Comment by nmrodelo** [comment]
@nikhilwoodruff is there anything left on my end for this PR?

**Comment by MaxGhenis** [comment]
Try renaming `snap_min_benefit` to `snap_minimum_benefit` in snap.py

File: openfisca_us/tests/policy/baseline/usda/snap/min_benefit.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  period: 2022
+  absolute_error_margin: 0.01
+  input:
+    people:
+      person_1:
+        age: 30
+      person_2:
+        age: 50
+      person_3:
+        age: 35
```
Comment (MaxGhenis) [inline openfisca_us/tests/policy/baseline/usda/snap/min_benefit.yaml:34]: This should fix the test
```suggestion
        age: 35
    spm_units:
      spm_unit:
        members: [person_1, person_2, person_3]
```

**Review by MaxGhenis — COMMENTED** [review]
Please also rename the test file to `snap_minimum_benefit.yaml` to match the variable name.

**Review by MaxGhenis — APPROVED** [review]

**Comment by MaxGhenis** [comment]
Thanks @nmrodelo, could you just do two more things? We're trying to get back on the http://semver.org track. You'll have to pull to fetch my latest PR.
1. Update the version to 0.4.0 in setup.py (this PR adds new functionality, so it's a minor change rather than patch)
2. Update CHANGELOG.md with an entry for 0.4.0 saying `Add SNAP minimum benefit`

I'll merge from there.

File: setup.py [hunk 5]
```diff
@@ -51,9 +51,7 @@
     },
     python_requires=">=3.7,<3.8",
     entry_points={
-        "console_scripts": [
-            "openfisca-us = openfisca_us.tools.cli:main",
-        ],
+        "console_scripts": ["openfisca-us = openfisca_us.tools.cli:main",],
```
Comment (MaxGhenis) [inline setup.py:54]: I think this will fix the lint
```suggestion
        "console_scripts": ["openfisca-us = openfisca_us.tools.cli:main",]
```

**Review by nikhilwoodruff — APPROVED** [review]
Looks great, thanks @nmrodelo. The semantic versioning GH action now looks like it's working (see the error message) - could you just add to the `CHANGELOG.md`? Looks good to merge straight after that

**Comment by nikhilwoodruff** [comment]
@nmrodelo thanks for this PR! Just added the changelog entry, merging.

**Pull request merged**